src/middlewares/notFoundMiddleware.ts

```diff
@@ -0,0 +1,7 @@
+import { Request, Response } from 'express';
+
+const notFoundMiddleware = (req: Request, res: Response) => {
+  res.status(404).json({ error: 'Not Found' });
+};
```

Comment:
This middleware will always response with 404 status due to missing condition inside and lack of next() function
https://expressjs.com/en/guide/using-middleware.html

Reply:
Could you please clarify what condition do you mean? And do I really need to call next() here?
As I understand from the expressjs articles - 404 is processed not as an error, so I thought I do not need to add additional checks and the next() call. Please correct me if I'm wrong.

Reply:
Yes, but in what case you will throw 404 error?
It's must be if something not found ...
But in your case it will throw in all situations when you will apply middleware

Reply:
It seems I didn't catch how to implement 404 correctly. Could you please share with me some articles regarding this topic?

Reply:
Could you answer my question, please?